internal/service/networkpeering/resource_network_peering_test.go

```diff
@@ -12,60 +12,16 @@ import (
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/mig"
 	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
 func TestAccNetworkNetworkPeering_basicAWS(t *testing.T) {
-	var (
-		peer                 matlas.Peer
-		resourceName         = "mongodbatlas_network_peering.test"
-		dataSourceName       = "data.mongodbatlas_network_peering.test"
-		pluralDataSourceName = "data.mongodbatlas_network_peerings.test"
-		orgID                = os.Getenv("MONGODB_ATLAS_ORG_ID")
-		vpcID                = os.Getenv("AWS_VPC_ID")
-		vpcCIDRBlock         = os.Getenv("AWS_VPC_CIDR_BLOCK")
-		awsAccountID         = os.Getenv("AWS_ACCOUNT_ID")
-		awsRegion            = os.Getenv("AWS_REGION")
-		providerName         = "AWS"
-		projectName          = acc.RandomProjectName()
-	)
-
-	resource.Test(t, resource.TestCase{
-		PreCheck:                 func() { acc.PreCheckBasic(t); acc.PreCheckPeeringEnvAWS(t) },
-		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
-		CheckDestroy:             acc.CheckDestroyNetworkPeering,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccMongoDBAtlasNetworkPeeringConfigAWS(orgID, projectName, providerName, vpcID, awsAccountID, vpcCIDRBlock, awsRegion),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckMongoDBAtlasNetworkPeeringExists(resourceName, &peer),
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttrSet(resourceName, "container_id"),
-					resource.TestCheckResourceAttr(resourceName, "provider_name", providerName),
-					resource.TestCheckResourceAttr(resourceName, "vpc_id", vpcID),
-					resource.TestCheckResourceAttr(resourceName, "aws_account_id", awsAccountID),
-
-					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
-					resource.TestCheckResourceAttrSet(dataSourceName, "container_id"),
-					resource.TestCheckResourceAttr(dataSourceName, "provider_name", providerName),
-					resource.TestCheckResourceAttr(dataSourceName, "vpc_id", vpcID),
-					resource.TestCheckResourceAttr(dataSourceName, "aws_account_id", awsAccountID),
+	resource.ParallelTest(t, *basicAWSTestCase(t))
+}
 
-					resource.TestCheckResourceAttrSet(pluralDataSourceName, "results.#"),
-					resource.TestCheckResourceAttrSet(pluralDataSourceName, "results.0.provider_name"),
-					resource.TestCheckResourceAttrSet(pluralDataSourceName, "results.0.vpc_id"),
-					resource.TestCheckResourceAttrSet(pluralDataSourceName, "results.0.aws_account_id"),
-				),
-			},
-			{
-				ResourceName:            resourceName,
-				ImportStateIdFunc:       testAccCheckMongoDBAtlasNetworkPeeringImportStateIDFunc(resourceName),
-				ImportState:             true,
-				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"accepter_region_name", "container_id"},
-			},
-		},
-	})
+func TestMigNetworkNetworkPeering_basicAWS(t *testing.T) {
```

Comment:
Choosing to put the migration test next to the acceptance test as I think the new `mig.CreateAndRunTest` makes it overkill to have a separate file.
But could go back to convention of a separate file?

Reply:
No strong opinion here but probably we should think about it because we recently decided to put resource and data source tests in the same test file. Having also migration tests in the same file could be inconvenient. It could be a good topic to bring to the technical discussion of tomorrow.

Reply:
I would favor sticking to existing convention for now for consistency. Feel free to bring up the topic with pros/cons and we can take decision from there.

Reply:
Moved in [1015e62](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2094/commits/1015e6206d7addfb2561ed7e4c9bcf6b14fddfb4)